homeassistant/components/mqtt/discovery.py

```diff
@@ -86,6 +89,17 @@ class MQTTConfig(dict):
     """Dummy class to allow adding attributes."""
 
 
+class MqttServiceInfo(TypedDict):
+    """Prepared info from mqtt entries."""
+
+    topic: str
+    payload: ReceivePayloadType
+    qos: int
+    retain: bool
+    subscribed_topic: str
+    timestamp: dt.datetime
+
+
```

Comment:
@emontnemery since mqtt is not strongly typed, I based this on:
https://github.com/home-assistant/core/blob/4f01631bd65a00ad1c4a1c5fe086038a13a97239/homeassistant/components/mqtt/models.py#L24-L32

Should `subscribed_topic` and `timestamp` be optional in `MqttServiceInfo`?
```python
    subscribed_topic: str | None
    timestamp: dt.datetime | None
```

Reply:
Great